File: src/test/java/com/miracoly/gameoflife/TestData.java
package com.miracoly.gameoflife;

import java.util.List;
import java.util.function.Supplier;

public class TestData {

    private TestData() {
    }


    /*
        x - - x -
        x - - x -
        - - - - -
        - - x - -
        - x - - -
     */
    public static final Supplier<Game> INITIAL_GAME_UNDERPOPULATION_ALL_DIE = () -> new Game(List.of(
            new Position(1, 1),
            new Position(4, 1),
            new Position(1, 2),
            new Position(4, 2),
            new Position(3, 4),
            new Position(2, 5)
    ));

    /*
    - - - - -
    - x x - -
    x - - x -
    - x x - -
    - - - - -
    */
    public static final Supplier<Game> INITIAL_STATIC_GAME = () -> new Game(List.of(
            new Position(2, 2),
            new Position(3, 2),
            new Position(1, 3),
            new Position(4, 3),
            new Position(2, 4),
            new Position(3, 4)
    ));

    /*
    - - - - -
    - - x - -
    - x x x -
    - - x - -
    - - - - -
    */
    public static final Supplier<Game> INITIAL_GAME_OVERPOPULATION = () -> new Game(List.of(
            new Position(3, 2),
            new Position(2, 3),
            new Position(3, 3),
            new Position(4, 3),
            new Position(3, 4)
    ));

    /*
    - - - - -
    - x x x -
    - x - x -
    - x x x -
    - - - - -
    */
    public static final Supplier<Game> NEXT_TICK_OVERPOPULATION = () -> new Game(List.of(
            new Position(3, 2),
            new Position(2, 3),
            new Position(4, 3),
            new Position(3, 4),
            new Position(2, 2),
            new Position(4, 2),
            new Position(2, 4),
            new Position(4, 4)
    ));

    /*
    - - - - -
    - - x - -
    - x - - -
    - - - x -
    - - - - -
    */
    public static final Supplier<Game> INITIAL_GAME_BIRTH = () -> new Game(List.of(
            new Position(3, 2),
            new Position(2, 3),
            new Position(4, 4)
    ));

    /*
    - - - - -
    - - - - -
    - - x - -
    - - - - -
    - - - - -
    */
    public static final Supplier<Game> NEXT_TICK_BIRTH = () -> new Game(List.of(
            new Position(3, 3)
    ));

}

File: haskell/test/TestData.hs
module TestData (module TestData) where

import GameOfLife (Board, Position (..))

{-
    x - - x -
    x - - x -
    - - - - -
    - - x - -
    - x - - -
-}
underpopulation :: Board
underpopulation =
  [ Position 1 1,
    Position 1 2,
    Position 2 5,
    Position 3 4,
    Position 4 1,
    Position 4 2
  ]

{-
    - - - - -
    - x x - -
    x - - x -
    - x x - -
    - - - - -
-}
static :: Board
static =
  [ Position 1 3,
    Position 2 2,
    Position 2 4,
    Position 3 2,
    Position 3 4,
    Position 4 3
  ]

{-
    - - - - -
    - - x - -
    - x x x -
    - - x - -
    - - - - -
-}
overpopulation :: Board
overpopulation =
  [ Position 2 3,
    Position 3 2,
    Position 3 3,
    Position 3 4,
    Position 4 3
  ]

{-
    - - - - -
    - x x x -
    - x - x -
    - x x x -
    - - - - -
-}
nextTickOverpopulation :: Board
nextTickOverpopulation =
  [ Position 2 2,
    Position 2 3,
    Position 2 4,
    Position 3 2,
    Position 3 4,
    Position 4 2,
    Position 4 3,
    Position 4 4
  ]

{-
    - - b - -
    - s d s -
    b d - d b
    - s d s -
    - - b - -
-}
secondNextTickOverpopulation :: Board
secondNextTickOverpopulation =
  [ Position 1 3,
    Position 2 2,
    Position 2 4,
    Position 3 1,
    Position 3 5,
    Position 4 2,
    Position 4 4,
    Position 5 3
  ]

{-
    - - - - -
    - - x - -
    - x - - -
    - - - x -
    - - - - -
-}
birth :: Board
birth =
  [ Position 2 3,
    Position 3 2,
    Position 4 4
  ]

{-
    - - - - -
    - - - - -
    - - x - -
    - - - - -
    - - - - -
-}
nextTickBirth :: Board
nextTickBirth = [Position 3 3]
